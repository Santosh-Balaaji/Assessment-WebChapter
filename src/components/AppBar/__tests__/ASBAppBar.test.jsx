import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import AppBarDrawerComponent from '../ASBAppBar';

describe('AppBar Component', () => {
  let wrapper;
  const mockHandleOnClose = jest.fn();
  const mockHandleOnOpen = jest.fn();
  const stubProps = {
    open: false,
    onClose: mockHandleOnClose,
    onOpen: mockHandleOnOpen,
  };

  beforeAll(() => {
    wrapper = shallow(<AppBarDrawerComponent {...stubProps} />);
  });

  it('should render default component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleOpen function on click of Burger Icon', () => {
    wrapper
      .find(
        'WithStyles(ForwardRef(IconButton))[data-testid="BurgerIconButton"]',
      )
      .simulate('click');
    expect(mockHandleOnOpen).toHaveBeenCalled();
  });

  it('should call handleClose function on click of Back Icon', () => {
    wrapper
      .find(
        'WithStyles(ForwardRef(IconButton))[data-testid="LeftArrorIconButton"]',
      )
      .simulate('click');
    expect(mockHandleOnClose).toHaveBeenCalled();
  });
});
