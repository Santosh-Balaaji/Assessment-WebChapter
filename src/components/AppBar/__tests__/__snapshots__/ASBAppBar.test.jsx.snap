// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Component should render default component 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(CssBaseline) />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-2"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="open drawer"
        className="makeStyles-menuButton-4"
        color="inherit"
        data-testid="BurgerIconButton"
        edge="start"
        onClick={[MockFunction]}
      >
        <Memo(ForwardRef(MenuIcon)) />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        noWrap={true}
        variant="h6"
      >
        Register Card Form
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    className="makeStyles-drawer-6"
    classes={
      Object {
        "paper": "makeStyles-drawerPaper-7",
      }
    }
    data-testid="Drawer"
    open={false}
    variant="persistent"
  >
    <div
      className="makeStyles-drawerHeader-8"
    >
      Menu
      <WithStyles(ForwardRef(IconButton))
        data-testid="LeftArrorIconButton"
        onClick={[MockFunction]}
      >
        <Memo(ForwardRef(ChevronLeftIcon)) />
      </WithStyles(ForwardRef(IconButton))>
    </div>
    <WithStyles(ForwardRef(Divider)) />
    <div>
      This is Menu Item
    </div>
  </WithStyles(ForwardRef(Drawer))>
</div>
`;
